### PR TITLE
Calendar backend

### PR DESCRIPTION
**Pull request by lubko43**

### CODE REVIEW

File: events/calendars/models.py [hunk 1]
```diff
@@ -0,0 +1,5 @@
+from __future__ import unicode_literals
```
Comment (lhalam) [inline events/calendars/models.py]: No need in this file. Please delete

File: events/calendars/views.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+from django.shortcuts import render
+from django.views.generic.base import View
+from django.http import JsonResponse
+import json
+from event.models import User, EventUserAssignment, Event
+
+# Create your views here.
+class Calendar(View):
+    """
+    View used .
```
Comment (lhalam) [inline events/calendars/views.py]: What?

File: events/calendars/views.py [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    """
+    def get(self, request, user_id=None):
+        if not user_id:
```
Comment (lhalam) [inline events/calendars/views.py]: You need to get user from request. Please do not pass user_id as param. Modify url pattern as well

File: events/calendars/views.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+            events = Event.objects.all()
```
Comment (lhalam) [inline events/calendars/views.py]: You need to get only events from specific account (you need to get account from user)

File: events/calendars/views.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+            result = [i.serialize() for i in events]
+            return JsonResponse({'all_events': result}, status=200)
```
Comment (lhalam) [inline events/calendars/views.py]: No need to pass ALL events fields - we just need event date and title

File: events/event/models.py [hunk 6]
```diff
@@ -44,3 +54,15 @@ def to_dict(self):
 class EventUserAssignment(models.Model):
     event = models.ForeignKey(Event, on_delete=models.CASCADE)
     user = models.ForeignKey(User, on_delete=models.CASCADE)
+
+    @staticmethod
+    def get_by_user_id(user_id):
+        try:
+            events = EventUserAssignment.objects.filter(user=user_id)
+            print len(events)
```
Comment (lhalam) [inline events/event/models.py]: Please drop

File: events/event/models.py [hunk 7]
```diff
@@ -44,3 +54,15 @@ def to_dict(self):
...
+            user_events = []
+            for event in events:
+                    user_events.append(event.event)
```
Comment (lhalam) [inline events/event/models.py]: Please check indents

File: events/event/models.py [hunk 8]
```diff
@@ -44,3 +54,15 @@ def to_dict(self):
...
+            return user_events
+        except Event.DoesNotExist:
```
Comment (lhalam) [inline events/event/models.py]: Did you check if this exception can be really raised here?

File: events/static/js/app/navbar.js [hunk 9]
```diff
@@ -0,0 +1,32 @@
...
+                        </ul>
+                        <ul className="nav navbar-nav navbar-right">
+                            <li><Link to="/auth/login?next=/" >Profile</Link></li>
+                        </ul>
+                    </div>
+                </nav>
+            </header>
+        );
+  }
+}
```
Comment (lhalam) [inline events/static/js/app/navbar.js]: Add empty line here

File: events/static/js/calendar1.js [hunk 10]
```diff
@@ -0,0 +1,52 @@
+import React from 'react'
+import moment from 'moment'
+import fullCalendar from 'fullcalendar'
+
+export default class Calendar extends React.Component {
+      render() {
```
Comment (lhalam) [inline events/static/js/calendar1.js]: Move to the bottom (render should be always last one)

File: events/static/js/calendar1.js [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+        return <div id="calendar"></div>;
+      }
+      componentDidMount() {
+        $('#calendar').fullCalendar({
```
Comment (lhalam) [inline events/static/js/calendar1.js]: jQuery code inside React app looks weird. Did you check any alternatives?

File: events/static/js/router.js [hunk 12]
```diff
@@ -0,0 +1,19 @@
+import React from 'react'
+import { render } from 'react-dom'
+import { Router, Route, Link, hashHistory } from 'react-router'
+
+import App from './app'
+import Calendar from './calendar1'
```
Comment (lhalam) [inline events/static/js/router.js]: What is it "calendar1"? Please do not use magic numbers.

File: events/registration/views.py [hunk 13]
```diff
@@ -13,7 +13,7 @@
 from events import settings
 
 
-CONFIRM_LINK = settings.HOST_NAME + '/api/v1/reg/confirm/'
+# CONFIRM_LINK = settings.HOST_NAME + '/api/v1/reg/confirm/'
```
Comment (BorysYudin) [inline events/registration/views.py]: Please uncomment this line

File: events/static/js/app/events.js [hunk 14]
```diff
@@ -0,0 +1,11 @@
...
+
+export default class Events extends React.Component {
+    render() {
+        return (
+            <div>
+                <h1>Events</h1>
+            </div>
+        );
+    }
+}
```
Comment (lhalam) [inline events/static/js/app/events.js]: Add empty line here

File: events/static/js/app/calendar.js [hunk 15]
```diff
@@ -0,0 +1,52 @@
...
+                        // if so, remove the element from the "Draggable Events" list
+                        $(this).remove();
+                    }
+                }
+        })
+      }
+      render() {
+        return <div id="calendar"></div>;
+      }
+}
```
Comment (lhalam) [inline events/static/js/app/calendar.js]: Add empty line here

File: events/static/js/app/navbar.js [hunk 16]
```diff
@@ -0,0 +1,32 @@
...
+                            <span className="sr-only">Toggle navigation</span>
+                            <span className="icon-bar"></span>
+                            <span className="icon-bar"></span>
+                            <span className="icon-bar"></span>
+                        </button>
+                        <Link to="/" className="navbar-brand" >Logo</Link>
+                    </div>
+                    <div id="navbarCollapse" className="collapse navbar-collapse">
+                        <ul className="nav navbar-nav">
+                            <li ><Link to="/">Home</Link></li>
```
Comment (lhalam) [inline events/static/js/app/navbar.js]: delete space

File: webpack.config.js [hunk 17]
```diff
@@ -0,0 +1,56 @@
...
+        ]
+    },
+
+    resolve: {
+        //tells webpack where to look for modules
+        modulesDirectories: ['node_modules'],
+        //extensions that should be used to resolve modules
+        extensions: ['', '.js', '.jsx']
+    }
+}
```
Comment (lhalam) [inline webpack.config.js]: add empty line

**Review by Scandie — APPROVED** [review]

**Pull request merged**